appsdk/sdk.go

```diff
@@ -380,6 +381,10 @@ func (sdk *AppFunctionsSDK) Initialize() error {
 			return fmt.Errorf("core service's version is not compatible with SDK's version")
 		}
 
+		if err := sdk.initializeSecretProvider(); err != nil {
```

Comment:
I think we need a flag indicating the SecretProvider was initialized so we don't tray again if DB below fails.